e2e/cypress/integration/messaging/collapse_link_spec.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Messaging', () => {
+    before(() => {
+        // # Set the configuration on Link Previews
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                EnableLinkPreviews: true,
+            },
+        });
+
+        cy.apiLogin('user-1');
+        cy.apiSaveLinkPreviewPreference(JSON.stringify(true));
+
+        cy.apiLogin('sysadmin');
+        cy.apiSaveLinkPreviewPreference(JSON.stringify(true));
+
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18708-Link preview - Removing it from my view removes it from other user\'s view', () => {
+        const message = 'https://www.bbc.com/news/uk-wales-45142614';
+
+        // # Create new DM channel with user's email
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Post message to use
+        cy.postMessage(message);
+
+        cy.getLastPostId().then((postId) => {
+            // * Check the preview is shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('exist');
+
+            // # Log-in to the other user
+            cy.apiLogin('sysadmin');
+            cy.visit('/ad-1/channels/town-square');
+
+            // * Check the preview is shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('exist');
+
+            // # Log-in back to the first user
+            cy.apiLogin('user-1');
+            cy.visit('/ad-1/channels/town-square');
+
+            // # Collapse the preview
+            cy.get(`#${postId}_message`).find('.post__embed-visibility').click({force: true});
+
+            // * Check the preview is not shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('not.exist');
+
+            // # Log-in to the other user
+            cy.apiLogin('sysadmin');
+            cy.visit('/ad-1/channels/town-square');
+
+            // * Check the preview is shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('exist');
+
+            // # Log-in back to the first user
+            cy.apiLogin('user-1');
+            cy.visit('/ad-1/channels/town-square');
+
+            // * Check the preview is still not shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('not.exist');
```

Comment:
When a user logs in, the file preview would be displayed. The assertion is wrong and hence the test fails at this step.

Reply:
Just to make sure I understood correctly. The collapse status of the links is not persistent between the different sessions of the same user?
Because if it is, when you log back in, the link should be the same way you left it. The test definition made me think this was the expected behaviour, but the test definition was considering two different browsers and less signing in and signing out.

Reply:
The image collapse state is not persistent into the server but to browser's localforage.  The test definition is based from 2 users who are logged in to 2 different browsers which we can't do in Cypress.
Normally on logout, those states are removed (but I found this inconsistent with Cypress).

With that, I'm with @srkgupta.  It's fine to remove above assertion.  Same case to **skip** step number 6:
```
6. Have the user who posted it click the arrows to re-expand the link preview
```